### PR TITLE
feat(filter-metadata): Filtered google lib metadata

### PR DESCRIPTION
**Pull request by nutboltu**
This PR excluded following metadata to reduce the bundle size. 
It's unlikely to use these meta.

1. noInternationalDialling
2. fixedLine
3. uan
4. tollFree
5. premiumRate
6. sharedCost
7.  voip
8. personalNumber
9. pager
10. voicemail

Also minified the dist/libphonenumber.js file

### CODE REVIEW

File: scripts/build-dependencies.sh [hunk 1]
```diff
@@ -1,3 +1,8 @@
+if [ ! -d ./libphonenumber/.git ]; then
```
Comment (superhit0) [inline scripts/build-dependencies.sh]: this means if the libphonenumber folder is not cached/corrupted, download from git,

we should not do this, we should always download libphonenumber

File: .travis.yml [hunk 2]
```diff
@@ -13,7 +13,6 @@ before_install:
 install:
   - npm ci
   - bash ./scripts/build-dependencies.sh
-  - git clone --branch v`cat package.json | python -c "import sys, json; print(json.load(sys.stdin)['googleLibphonenumberVersion'])"` https://github.com/googlei18n/libphonenumber/
```
Comment (superhit0) [inline .travis.yml]: i would not suggest not removing this

**Review by superhit0 — DISMISSED** [review]
will be needing more time to review further.

I have given some initial thoughts

**Comment by superhit0** [comment]
@nutboltu can you also remove the sub module `python-gflags` as well?

its not required for development

**Comment by superhit0** [comment]
Lets look at the diff of the our generated files & this https://github.com/googlei18n/libphonenumber/blob/master/javascript/i18n/phonenumbers/metadatalite.js

**Comment by superhit0** [comment]
@nutboltu 
there is a problem with removing these metadatas:: the isvalidnumber starts giving wrong answers, for example we removed fixed line, then I tested a fixed line number from India, it said invalid, even though it was valid.

Local:
image

Live:
image

**Comment by nutboltu** [comment]
@superhit0 This is the starting point. I think we need to list down the number type and other metadata which  we will remove from the validation.

I tried adding `fixed line` and it works fine for your given example. In a broader view. we can remove
`uan`, `tollFree`, `premiumRate`, `sharedCost`, `voip`, `personalNumber`, `pager`, `voicemail` and also `leadingDigits`.

File: .travis.yml (hunk 2 above)
Comment (nutboltu) [reply .travis.yml]: As I moved this line to `./scripts/build-dependencies.sh`. Do we need to add it again in travis.yml?

Comment (superhit0) [reply .travis.yml]: nah, its okay

**Comment by superhit0** [comment]
Also, we need to strengthen the tests in the repo, before we merge this one, I will try adding tests with different types of phonenumber as well.

@nutboltu @patw0929 I would request you guys to do the same as well please.

**Comment by nutboltu** [comment]
@superhit0 I tested it in `react-intl-tel-input`. Now it only supports `mobile` and `fixed line`. Is it okay not to support rest?

**Comment by superhit0** [comment]
@nutboltu whats the file size now?

**Comment by nutboltu** [comment]
@superhit0 Currect file size is `~179KB`

**Comment by nutboltu** [comment]
downgrading node version in CI. https://github.com/facebook/jest/issues/8069

**Comment by superhit0** [comment]
hmm for 40 KB should we remove logics for different types of phones?
@patw0929 @nutboltu what do you think?

**Comment by nutboltu** [comment]
If we also remove `fixed line` it will down to `~139KB`. It upon us what we want to keep and we don't.

**Comment by superhit0** [comment]
Thats the thing right ?
we cannot remove fixed line as most of the offices have fixed line numbers

**Comment by nutboltu** [comment]
@patw0929 , @superhit0 
I would like to explain the facts. Previously our `react-intl-tel-input@5.0.1` had a bundle size of `72.7KB`. But now after incorporating this `libphonenumber-js-utils`, the size increases to `286.9KB`.

This fact is we are only using following functions from this library.

- formatNumber
- getExampleNumber
- getExtension
- getNumberType
- isValidNumber

But there's no tree shaking in the bundle. Considering those functions, I would say we can keep
- formatNumber
- getExtension
- getNumberType

I am not sure what's the benefits of keeping `getExampleNumber`. Giving an example provided by google may not necessary. And this may reduce the bundle size.

Now the main reason behind this huge bundle size is this `isValidNumber` function. I can say every developer validates the phone number in the backend as well and keeping this validation in the client side caused `~160KB` size data. Still if we want to keep the validation, instead of using this google compiler I would suggest to build our own compiler using google metadata. (e.g: [libphonenumber-js](https://github.com/catamphetamine/libphonenumber-js)).

There's few possible solution I can say:

1.  Build our own compiler using google metadata.
2. Use [libphonenumber-js](https://github.com/catamphetamine/libphonenumber-js). [react-phone-number-input](https://github.com/catamphetamine/react-phone-number-input) is using this library.
3. Reduce the supported number type. Because very few people are using the following type of number.
   - noInternationalDialling
   - uan
   - tollFree
   - premiumRate
   - sharedCost
   - voip
   - personalNumber
   - pager
   - voicemail
We can only support `mobile` and `fixedLine` for now which will reduce `~40 KB`. It's really a down side when we load this component in a slow internet connection. We should consider it seriously.

**Comment by nutboltu** [comment]
@patw0929 any thoughts about this PR?

**Comment by patw0929** [comment]
Sweet! 🙌 I agree with you that we should consider reducing the bundle size.

Just curious, what's the difference between the three solutions you listed above?
If we choose to use libphonenumber-js instead, is the approach also support mobile & fixed line only? And how big is the bundle size?
We need more information to make a decision.

**Comment by nutboltu** [comment]
@patw0929 
For the 1st solution we need to build our own compiler using google metadata. In this case, we can keep updated with google metadata like `libphonenumber-js`. It's kind of rebuilding `libphonenumber-js` in our own way.

For the 2nd solution, it's a dependency to this package.
They only support `mobile`. see this line
https://github.com/catamphetamine/libphonenumber-js/blob/5678d38e64d314d75dad994f67fd833f347551eb/source/tools/generate.js#L35
Their bundle size is 27.9Kb (https://bundlephobia.com/result?p=libphonenumber-js@0.2.2)

3rd solution is this PR where we specify the supported number type.

**Comment by patw0929** [comment]
Thank you for your detailed explanation! 👍 

If we only want to support `mobile`, maybe we can just use `libphonenumber-js`, for convenience.

I personally think it is worthy to support mobile & fixed line.
So if you guys has same thought, then which solution is better for the maintenance? And pros & cons?
Or if we choose 3rd solution (this PR), what's next? Is it ready for merge?
Thanks!

**Comment by nutboltu** [comment]
@patw0929 This PR is to support only mobile and fixed line. Please review this so that we can merge

File: package.json [hunk 3]
```diff
@@ -9,10 +9,14 @@
...
+    "filterxml": "^1.1.4",
+    "node-minify": "^3.6.0"
+  },
   "devDependencies": {
     "chai": "^4.2.0",
-    "jsdom": "^13.1.0",
-    "jest": "^23.6.0"
+    "jest": "^23.6.0",
+    "jsdom": "^13.1.0"
   },
```
[the remark this answers is not in the document]
Comment (nutboltu) [reply package.json]: I was wondering why package-lock wasn't in my git changes. fixed it

[the remark this answers is not in the document]
Comment (patw0929) [reply package.json]: Thank you!
But it is failed in CI, can you help to solve it?

[the remark this answers is not in the document]
Comment (nutboltu) [reply package.json]: We need to change `oraclejdk8` to `openjdk8`. Now it's all good.

**Review by patw0929 — APPROVED** [review]
Looks great to me!

**Comment by nutboltu** [comment]
@patw0929 , @superhit0  are we happy to merge this PR?

**Comment by nutboltu** [comment]
@patw0929  I am going to merge this PR.

**Pull request merged**